### ISSUE TITLE
[Accessibility]: Disabled elements with `<Tooltip>` introduce `tabIndex` without a `role`

### ISSUE THREAD

**Issue by courier-new** [sourcegraph/sourcegraph#38782]
### Audit type

ARC Toolkit

### Problem description

When a disabled element is wrapped in a `<Tooltip />`, the trigger introduces an element to the tab order that lacks a `role`, causing an error to be reported by ARC. For example, with a disabled button:

<img width="1264" alt="image" src="https://user-images.githubusercontent.com/8942601/178863983-a1e902c1-3253-4118-900d-4c24e666045c.png">

### Expected behavior

ARC Toolkit keyboard checks pass.

### Additional details

I am unsure if introducing a role is the proper fix here and will leave for FPT to triage.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by felixfbecker** [comment]
See also https://github.com/sourcegraph/sourcegraph/issues/41683
